README.md

```diff
@@ -165,6 +165,11 @@ run('http-server .', {async: true}).then((output) => {
 }).catch((error) => {
   throw error
 })
+run('webpack', {asyncSpawn: true, args: ['--progress', '-p']}).then((output) => {
```

Comment:
Proposition for `runAsync`:
```javascript
function runAsync (command, options) {
  // Prepare options for exec command (don't need async and stdio as it doesn't handle them)
  const spawnOptions = Object.assign({shell: true}, options)
  const timeout = spawnOptions.timeout
  delete spawnOptions.async

  return new Promise((resolve, reject) => {
    const asyncProcess = childProcess.spawn(command, [], spawnOptions)
    asyncProcess.on('error', (error) => {
      reject(error)
    })

    asyncProcess.on('close', (exitCode) => {
      if (exitCode === 0) {
        resolve(exitCode)
      } else {
        reject(new Error(`Command failed: ${command}`))
      }
    })

    if (timeout) {
      setTimeout(() => {
        asyncProcess.kill()
        reject(new Error(`Command timeout: ${command}`))
      }, timeout)
    }
  })
}
```
Timeout is solved here, the only thing which is lacking is the outcome of the command in `resolve` under `close` event. We could return data from buffer there.